src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -41,18 +46,19 @@
 /**
  * Created on 8/10/16.
  */
-@Listeners(TestListener.class)
-public class Locomotive extends Watchman implements Conductor<Locomotive> {
+@Listeners({TestListener.class, SauceOnDemandTestListener.class})
+public class Locomotive extends Watchman implements Conductor<Locomotive>, SauceOnDemandSessionIdProvider, SauceOnDemandAuthenticationProvider {
 
     private static final float SWIPE_DISTANCE = 0.25f;
     private static final float SWIPE_DISTANCE_LONG = 0.50f;
     private static final float SWIPE_DISTANCE_SUPER_LONG = 1.0f;
     private static final int SWIPE_DURATION_MILLIS = 2000;
 
-    public ConductorConfig configuration;
-    public AppiumDriver driver;
```

Comment:
breaks backwards compatibility but a very nice move!

Reply:
It does, you have to in order to add thread safety...had to bite the bullet and do it for the Web version.